src/pocketbase/services/batch.py

```diff
@@ -0,0 +1,143 @@
+        self._requests: list[BatchRequest] = []
+        self._subs: dict[str, SubBatchService] = {}
+
+    # we can use it with `async with` syntax
+    async def __aenter__(self) -> "BatchService":
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc: BaseException | None,
+        tb: Any,
+    ) -> None:
+        return None
+
+    def collection(self, collection_name: str) -> "SubBatchService":
+        """
+        Begin building a batch for the given collection.
+
+        Returns a `SubBatchService` to queue CRUD operations.
+        """
+        if collection_name not in self._subs:
+            self._subs[collection_name] = SubBatchService(
+                self._requests,
+                collection_name,
+            )
+        return self._subs[collection_name]
+
+    async def send(self, options: CommonOptions | None = None) -> list[BatchRequestResult]:
+        send_opts: SendOptions = {
+            "method": "POST",
+        }
+
+        if options:
+            send_opts.update(options)
+
+        # building the request body
+        json_data = []
+        all_files: SendableFiles = []
+
+        for idx, req in enumerate(self._requests):
+            body_transformed, files = transform(req.get("body", {}))
+            json_data.append(
+                {
+                    "method": req["method"],
+                    "url": req["url"],
+                    "headers": req.get("headers"),
+                    "body": body_transformed,
+                }
+            )
+            if files:
+                for key, file in files:  # type: ignore
+                    updated_key = f"requests.{idx}.{key}"
+                    all_files.append((updated_key, file))
+
+        send_opts["body"] = {"@jsonPayload": json.dumps({"requests": json_data})}
+        if len(all_files) == 0:
+            # add dummy files to trigger multipart encoding
+            # easiest way to trigger multipart/form-data encoding
+            # this key will be ignored by PocketBase
+            # https://github.com/pocketbase/js-sdk/blob/b17fd45624a86d69b56163da6e8a519f533a1f22/src/services/BatchService.ts#L71
+            all_files.append(("dummy.txt", b"mamaistdiebeste"))
+        send_opts["files"] = all_files
+
+        result_request: list[JsonType] = await self._send("", send_opts)  #  type: ignore
+        result = []
+        for item in result_request:
+            result_item: BatchRequestResult = {
+                "status": item["status"],  #  type: ignore
+                "body": item.get("body", {}),  #  type: ignore
+            }
+            result.append(result_item)
+        self._requests.clear()
+        return result
+
+
+class SubBatchService:
+    """
+    Helper to queue CRUD ops for a specific collection into the batch.
+    """
+
+    def __init__(self, requests: list[BatchRequest], collection: str) -> None:
+        self._requests = requests
```

Comment:
I rather have you keep a reference to the BatchService then the list within the service and call self._batch_service._submit_request() or something.